android/src/main/java/expo/modules/blueskyvideo/FullscreenActivity.kt

```diff
@@ -25,37 +29,54 @@ class FullscreenActivity : AppCompatActivity() {
             return
         }
 
-        this.window.setFlags(
-            WindowManager.LayoutParams.FLAG_FULLSCREEN,
-            WindowManager.LayoutParams.FLAG_FULLSCREEN,
-        )
-
-        val keepDisplayOn = this.getIntent().getBooleanExtra("keepDisplayOn", false)
+        // Enable full-screen mode for older Android versions while avoiding double insets
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R) {
+            @Suppress("DEPRECATION")
+            window.decorView.systemUiVisibility = (
+                View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                        or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                        or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+            )
+        } else {
+            window.setDecorFitsSystemWindows(false)
+        }
 
+        // Keep screen on if requested
+        val keepDisplayOn = this.intent.getBooleanExtra("keepDisplayOn", false)
         if (keepDisplayOn) {
-            this.window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
+            window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
         }
 
-        // Update the player viewz
-        val playerView =
-            PlayerView(this).apply {
-                setBackgroundColor(Color.BLACK)
-                setShowSubtitleButton(true)
-                setShowNextButton(false)
-                setShowPreviousButton(false)
-                setFullscreenButtonClickListener {
-                    finish()
-                }
+        // Update the player view with conditional insets
+        val playerView = PlayerView(this).apply {
+            setBackgroundColor(Color.BLACK)
+            setShowSubtitleButton(true)
+            setShowNextButton(false)
+            setShowPreviousButton(false)
+            setFullscreenButtonClickListener {
+                finish()
+            }
+
+            layoutParams = ViewGroup.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT
+            )
+            useController = true
+            controllerAutoShow = false
+            controllerHideOnTouch = true
 
-                layoutParams =
-                    ViewGroup.LayoutParams(
-                        ViewGroup.LayoutParams.MATCH_PARENT,
-                        ViewGroup.LayoutParams.MATCH_PARENT,
-                    )
-                useController = true
-                controllerAutoShow = false
-                controllerHideOnTouch = true
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                // Apply padding dynamically for modern Android
+                setOnApplyWindowInsetsListener { _, insets ->
+                    val systemInsets = insets.getInsets(android.view.WindowInsets.Type.systemBars())
```

Comment:
What does this return on <14? Do we need to special case 15 or can we use the same logic, using the system insets as the baseline for all calculations?

Reply:
On old versions it results in double insets, since they don't use edge-to-edge mode. But maybe we could force e2e mode there? lemme see

Reply:
Yeah sorry don't mean to question the logic, assume you tried this too, just feels like we should be able to calculate a baseline for all platforms, and _then_ apply adjustments to work around any of our own custom UI.